jl4-lsp/src/LSP/L4/Actions.hs

```diff
@@ -96,6 +103,54 @@ gotoDefinition pos m positionMapping = do
   newRange <- toCurrentRange positionMapping lspRange
   pure (Location (fromNormalizedUri range.moduleUri) newRange)
 
+
+-- ----------------------------------------------------------------------------
+-- Ladder evalApp
+-- ----------------------------------------------------------------------------
+
+evalApp
+  :: MonadIO m
```

Comment:
you can scope your `m` here with `forall m.`, then you don't get a new `m` in each decl in the `where` (that should save you the constraints)

Reply:
Nice. I've managed to get the `m` in wide scope / to be the same throughout, but I don't know how to get the right type for `method` (it shouldn't be *any* method). 

```
evalApp
  :: forall m method.
  (MonadIO m)
  => TypeCheckResult
  -> Ladder.EvalAppRequestParams
  -> RecentlyVisualised
  -> EL.Environment
  -> ExceptT (TResponseError method) m Aeson.Value
```
Then again, it also isn't a big deal whether we get the `method` type param right